low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -128,6 +136,74 @@ def get_controllers_info(self):
             data.append(controller_dict)
         return data
 
+    def get_logical_volumes_info(self):
+        """Update and return logical volume information in specific format."""
+        logvol_data = []
+        logicalvolumes = self.get_realstor_encl_data("volumes")
+        if logicalvolumes:
+            for logicalvolume in logicalvolumes:
+                health = logicalvolume.get("health", "NA")
+                pool_sr_no = logicalvolume.get("container-serial", "NA")
+                uid = logicalvolume.get("volume-name", "NA")
+                if pool_sr_no in self.dg_vol_map:
+                    self.dg_vol_map[pool_sr_no].append({"volume_uid": uid})
+                else:
+                    self.dg_vol_map.update({pool_sr_no:[{"volume_uid": uid}]})
+                if health in self.undesired_vals:
+                    health = "NA"
```

Comment:
keep these desired & undesired values at one place & use, like in sspl-constants

Reply:
Okay. Done